powertools-parameters/src/test/java/software/amazon/lambda/powertools/parameters/DynamoDbProviderTest.java

```diff
@@ -0,0 +1,167 @@
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.*;
+import software.amazon.lambda.powertools.parameters.cache.CacheManager;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.MockitoAnnotations.openMocks;
+
+public class DynamoDbProviderTest {
+
+    @Mock
+    DynamoDbClient client;
+
+    @Captor
+    ArgumentCaptor<GetItemRequest> getItemValueCaptor;
+
+    @Captor
+    ArgumentCaptor<QueryRequest> queryRequestCaptor;
+
+
+    private DynamoDbProvider provider;
+    private final String tableName = "ddb-test-table";
+
+    @BeforeEach
+    public void init() {
+        openMocks(this);
+        CacheManager cacheManager = new CacheManager();
+        provider = new DynamoDbProvider(cacheManager, client, tableName);
+    }
+
+
+    @Test
+    public void getValue() {
+
+        // Arrange
+        String key = "Key1";
+        String expectedValue = "Value1";
+        HashMap<String, AttributeValue> responseData = new HashMap<String, AttributeValue>();
+        responseData.put("id", AttributeValue.fromS(key));
+        responseData.put("val", AttributeValue.fromS(expectedValue));
+        GetItemResponse response = GetItemResponse.builder()
+                .item(responseData)
+                .build();
+        Mockito.when(client.getItem(getItemValueCaptor.capture())).thenReturn(response);
+
+        // Act
+        String value = provider.getValue(key);
+
+        // Assert
+        assertThat(value).isEqualTo(expectedValue);
+        assertThat(getItemValueCaptor.getValue().tableName()).isEqualTo(tableName);
+        assertThat(getItemValueCaptor.getValue().key().get("id").s()).isEqualTo(key);
+    }
+
+
+    @Test
+    public void getValueWithoutResultsReturnsNull() {
+        // Arrange
+        Mockito.when(client.getItem(getItemValueCaptor.capture())).thenReturn(GetItemResponse.builder()
+                .item(null)
+                .build());
+
+        // Act
+        String value = provider.getValue("key");
+
+        // Assert
+        assertThat(value).isEqualTo(null);
+    }
+
+    @Test
+    public void getValueWithMalformedRowThrows() {
+        // Arrange
+        String key = "Key1";
+        HashMap<String, AttributeValue> responseData = new HashMap<String, AttributeValue>();
+        responseData.put("id", AttributeValue.fromS(key));
+        responseData.put("not-val", AttributeValue.fromS("something"));
+        Mockito.when(client.getItem(getItemValueCaptor.capture())).thenReturn(GetItemResponse.builder()
+                .item(responseData)
+                .build());
+        // Act
+        Assertions.assertThrows(NullPointerException.class, () -> {
```

Comment:
Are you sure it should be nullpointer? Couldn't we have a better (eventually custom) one?

Reply:
Throwing a `DynamoDbProviderSchemaException` now with the error text describing the missing column